ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/sequence/actions/FootstepPlanActionPlanningThread.java

```diff
@@ -91,6 +93,20 @@ else if (definition.getPlannerInitialStanceSide().getValue() == InitialStanceSid
 
       footstepPlanner.getFootstepPlannerParameters().set(definition.getPlannerParametersReadOnly());
 
+      // TODO: Add body path planning options to user
+      footstepPlannerRequest.setPlanBodyPath(false);
+      if (!footstepPlannerRequest.getPlanBodyPath())
+      {
+         // At beginning, first turn in place to face the direction that the goal stance faces
+         startMidFeetPose.interpolate(startFootPoses.get(RobotSide.LEFT), startFootPoses.get(RobotSide.RIGHT), 0.5);
+         goalMidFeetPose.interpolate(goalFootPoses.get(RobotSide.LEFT), goalFootPoses.get(RobotSide.RIGHT), 0.5);
+         startTurnedToMatchGoalFacing.set(startMidFeetPose);
+         startTurnedToMatchGoalFacing.getOrientation().set(goalMidFeetPose.getOrientation());
+         footstepPlannerRequest.getBodyPathWaypoints().add(startMidFeetPose);
+         footstepPlannerRequest.getBodyPathWaypoints().add(startTurnedToMatchGoalFacing);
+         footstepPlannerRequest.getBodyPathWaypoints().add(goalMidFeetPose);
+      }
```

Comment:
I added this block which makes sure when planning backwards, the robot doesn't try to turn around just to take a few steps back.

Reply:
The use turn walk turn checkbox literally is the setting for whether to use A*:
```
footstepPlannerRequest.setPerformAStarSearch(!definition.getPlannerUseTurnWalkTurn().getValue());
```

The turn walk turn planner however take way more unnecessary steps than A*, which is why it should be preferred. I'm gonna makes some changes to clear this up.